parser_library/src/workspaces/processor_file_impl.cpp

```diff
@@ -54,6 +61,7 @@ parse_result processor_file_impl::parse(parse_lib_provider& lib_provider,
         last_opencode_id_storage_ = std::make_shared<context::id_storage>();
 
     const bool collect_hl = should_collect_hl();
+    fade_messages_->clear();
```

Comment:
This will drop the fading if the parsing is cancelled. I think the face_messages_ should be shared_ptr to const, a new non-const should be allocated here and in parse_macro. And only after successful parsing the shared_ptrs should be swapped.